qcfractal/interface/client.py

```diff
@@ -4,15 +4,21 @@
 import os
 import re
 from collections import defaultdict
-from typing import Any, Dict, List, Optional, Union
+from typing import Any, DefaultDict, Dict, List, Optional, Tuple, Union
 
 import pandas as pd
 import requests
+
 from pydantic import ValidationError
+from qcfractal.interface.models import TaskRecord
 
 from .collections import collection_factory, collections_name_map
-from .models import GridOptimizationInput, Molecule, ObjectId, TorsionDriveInput, build_procedure
-from .models.rest_models import ComputeResponse, rest_model
+from .collections.collection import Collection
+from .models import (GridOptimizationInput, KeywordSet, Molecule, ObjectId, ResultRecord, TorsionDriveInput,
+                     build_procedure)
+from .models.rest_models import (ComputeResponse, KeywordGETResponse, MoleculeGETResponse, ProcedureGETResponse,
+                                 QueryObjectId, QueryProjection, QueryStr, ResultGETResponse, ServiceQueueGETResponse,
+                                 TaskQueueGETResponse, rest_model)
```

Comment:
What about something like: https://github.com/samuelcolvin/pydantic/blob/master/pydantic/main.py#L23?

Also please do `isort` this when you get the chance.

Reply:
Was not aware of `TYPE_CHECKING`, it makes a lot of sense, and I will update this PR accordingly.